docs/wiki/The CI environment - Pipeline design.md

```diff
@@ -0,0 +1,146 @@
+| **setEnvironmentVariables** | This action parses the variables file `global.variables.yml` and sets the key-value pairs in the `variables` list as environment variables. |
+
+Technical documentation for each composite action, such as required input and output variables, is included in each `action.yml` file located in path `.github/actions/templates`.
+
+</details>
+
+<details>
+<summary>Azure DevOps</summary>
+
+<img src=".\media\CIEnvironment\pipelinePhasesADO.png" alt="Pipeline phases ADO" height="300">
+
+Azure DevOps pipelines map each pipeline phase to a dedicated pipeline template, to maximize code reusability.
+The mapping to the specific yaml template file is provided below:
+
+| Template Name | Pipeline phase |
+| - | - |
+| **jobs.validateModulePester.yml** | Static validation |
+| **jobs.validateModuleDeployment.yml** | Deployment validation |
+| **jobs.publishModule.yml** | Publishing |
+
+Technical documentation for each template, such as required input and output variables, is included in each `.yml` file located in path `.azuredevops/pipelineTemplates`.
+
+</details>
+
+## Module pipeline inputs
+
+Each module pipeline comes with the following runtime parameters:
+
+- `'Branch' dropdown`: A dropdown to select the branch to run the pipeline from.
+- `'Remove deployed module' switch`: Can be enabled or disabled and controls whether the test-deployed resources are removed after testing. It is enabled by default.
+- `'Publish prerelease module' switch`: Can be enabled or disabled and allows publishing a prerelease version for the corresponding module when running the pipeline from a branch different than [main|master]. It is disabled by default. For further information of how the input is processed refer to the [Publishing](./The%20CI%20environment%20-%20Publishing) dedicated page.
+
+  <img src=".\media\CIEnvironment\modulePipelineInput.png" alt="Module Pipeline Input" height="300">
+
+---
+
+# Platform pipelines
+
+In addition to module pipelines, the repository includes several platform pipelines covering further tasks as described below.
+
+- [Dependencies pipeline](#dependencies-pipeline)
+- [ReadMe pipeline](#readme-pipeline)
+- [Wiki pipeline](#wiki-pipeline)
+
+## Dependencies pipeline
+
+In order to successfully run module pipelines to validate and publish CARML modules to the target environment, certain Azure resources need to be deployed beforehand.
+
+For example any instance of the [Virtual Machine] module needs an existing virtual network to be connected to and a key vault hosting its required local admin credentials to be referenced.
+
+The dependencies pipeline covers this requirement and is intended to be run before executing module pipelines successfully.
+
+You can find further details about the dependencies pipeline in the  [Getting started - Dependency pipeline](./Getting%20started%20-%20Dependency%20pipeline) section.
+
+### Dependencies pipeline inputs
```

Comment:
Should we move this into the [Getting started - Dependency pipeline](./Getting%20started%20-%20Dependency%20pipeline) section?

Reply:
The thought came up as I figured somebody processing the Getting-started may wonder about these inputs too

Reply:
I'm also in doubt on what to do with the dependency sections. We probably have too many details in the getting started which is intended to be a swift step by step guide on how to deploy dependencies. We should maybe revisit that offline a bit.

Reply:
Dependencies pipeline docs updates after syncing offline:
- `Getting started - Setup environment`, dependencies pipeline paragraph: contains link to pipeline usage 
- `CI environment - Pipeline usage`: contains step by step instructions on how to run the pipeline
- `CI environment - Pipeline design`: contains details on the design, e.g. deployed resources and key vault secrets and is referenced by both pages above

The `Getting started - dependency pipeline` page has been removed and its content included in the `CI environment - pipeline design` page in the `dependencies pipeline` section